### DISCUSSION TITLE
Custom configurations doesn't set permanently

### DISCUSSION THREAD

**Discussion by SarjyantShrestha** [prasanthrangan/hyprdots#913] [Q&A] [answered]
### Description

Every time I customize the settings to my preference, they reset whenever I switch themes or even just change the wallpaper.
I'm unsure whether this issue is a genuine bug or simply due to my limited understanding of Linux configurations. If it's due to my lack of understanding, then I apologize and would greatly appreciate any guidance.

### Steps to Reproduce

1. Customize configuration (eg: remove animation from waybar)
2. Change theme or wallpaper using default shortcuts

### Expected Behavior

Set the configurations I customized permanently, if not for all themes then atleast for the current theme I'm using.

### Actual Behavior

Configurations seems to reset to default when I change themes.

**Comment by kRHYME7** [comment]
Hello!  can you list down all the files you want to modify? 


I will move this to discussions panel as this is not an issue.

**Reply by SarjyantShrestha** [reply]
> Hello! can you list down all the files you want to modify?
> 
waybar, hyprland, kitty.. can I say everything? Because I'm kinda new to linux configuration and I don't know what I will want to configure in the future.

**Reply by kRHYME7** [reply, marked as answer]
hmmm , you know symlinks? 


in kitty
each time you edit the theme.conf you are editing the original file. 
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/85618479-b8d8-461e-869a-946abfd9f842)

in hyprland same goes to kitty. 
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/b77e8773-5b32-4320-99b5-4d6a262efc5a)

in waybar that's a symlink too
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/0a5b6cd0-e442-41c5-ac12-a9887d22d552)
Don't ever try to change something inside this as this will be overwritten
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/25150490-cbde-4fd5-8382-95b8259a0ce9)

And many more, can I say everything ? 

Now, every time you change themes/wallpaper, that symbolic link will change according to the current theme. Now you got answers why it changed. 
If you want to have specific changes for every theme, then you should edit all.


Read this theming guide for further info, see how the theming works. 
https://github.com/prasanthrangan/hyprdots/wiki/Theming
 
If you plan to make a custom theme, we can help with that, as Hyprdots have an ability to patch someone's theme 😁 Thanks for sharing in advance. 
Good luck!

**Reply by SarjyantShrestha** [reply]
Thank you for your detailed reply.

**Reply by SarjyantShrestha** [reply]
Hey. I have a question!
If I want to copy my custom hyprdots configurations to a freshly installed Arch, what would be the proper way to do it?
Do I first install Hyprdots using the script and then overwrite my custom configurations?

**Reply by kRHYME7** [reply]
> Do I first install Hyprdots using the script and then overwrite my custom configurations?

I recommend using this. 

Or if you have a fork of this in GitHub use that fork for installation.


Another way without learning Git is to run ``` Hyprdots backup all ``` this will just back up all configs to this path ```  ~/.config/cfg_backups/* ``` . Copy that configs to your flash drive, then on a separate computer install the proper hyprdots the normal way. After that move, the copied backup files to the new computer then run ``` Hyprdots restore BackUp /path/to/the/backup ```  it will restore those .configs for you. 
System files are not included (sddm grub etc) 


I recommend managing it using git, it will be easier.

**Reply by nashvillehotchicken** [reply]
Hello, sorry for the necro-post. I'm stupid and I don't know how to stop config.jsonc from being overwritten. Is there a guide somewhere? Thanks. All I need to do is keep 1 line.


    "layer": "top",
    "output": "DP-1" <-- this one
    "position": "top",
    "mod": "dock",
    "height": 28,
    "exclusive": true,
    "passthrough": false,
    "gtk-layer-shell": true,
    "reload_style_on_change": true,


The line I want to keep is `"output": "DP-1",`
It is the only line I've added and it's pretty annoying to re-add it every time I change the theme.

Thank you.

Sorry for being stupid.

EDIT: vvvvv

Clarification:
What file do I actually edit for the changes to be permanent and not be affected by theme switches?

**Reply by kRHYME7** [reply]
this thread is outdated, so information here is not helpful.


 edit the modules instead 

`~/.config/waybar/modules/header.jsonc ` 

another trick is to make a new file 

` ~/.config/waybar/config` then add your custom in here and just source the dynamically genertaed config.jsonc

**Reply by nashvillehotchicken** [reply]
> edit the modules instead
> 
> `~/.config/waybar/modules/header.jsonc `
> 
> another trick is to make a new file
> 
> ` ~/.config/waybar/config` then add your custom in here and just source the dynamically genertaed config.jsonc

thank you!

**Comment by se7en-x230** [comment]
I think it would be a good idea to write a short wiki because I think many user (including me) struggle to understand what the best procedure is to keep modified settings in hyprland/dunst/waybar after an update.

A struictured step by step guide what to do to keep certain settings untouched and what steps to run to update the rest of the repo

**Reply by SarjyantShrestha** [reply]
> I think it would be a good idea to write a short wiki because I think many user (including me) struggle to understand what the best procedure is to keep modified settings in hyprland/dunst/waybar after an update.

It changes even after switching themes (Not just after an update).

**Reply by kRHYME7** [reply]
Feel free to push an MR regarding the documentation, will gladly help in implementing.
